src/util/translation.h

```diff
@@ -47,8 +48,30 @@ inline bilingual_str operator+(bilingual_str lhs, const bilingual_str& rhs)
     return lhs;
 }
 
+namespace util {
+/**
+ * Translation function.
+ * If no translation function is set, simply return the input.
+ */
+inline std::string translate(const char* lit)
+{
+    return G_TRANSLATION_FUN ? G_TRANSLATION_FUN(lit) : lit;
+}
+/** Type to denote whether an original string literal is translatable */
+template <bool translatable = true>
+struct Translatable {
+    const char* const lit;
+    consteval Translatable(const char* str) : lit{str} { assert(lit); }
+    std::string translate() const { return translatable ? util::translate(lit) : lit; }
+    operator bilingual_str() const { return {lit, translate()}; }
+};
+} // namespace util
+
+consteval auto _(util::Translatable<true> str) { return str; }
```

Comment:
In commit "refactor: Delay translation of Untranslated() or _() literals" (fa62737ae398bd56ef9e68b8d0a4d217cfcc532b)

Thinking about this more it seems like if `_` function does not be consteval, and could just a template type that inherits from `bilingual_str`, and included the number of format specifiers included in the string in its type information. This would be nice because runtime semantics of `_` would be unchanged, it would still do the translation at time it is called, and there would be no need to make any `translated` -> `translate()` replacements. 

I think the main downside to this approach is would be need to refactor CheckNumFormatSpecifiers but there would be fewer changes overall. The other downside would be that _ would need to be aware the translated strings might be a format string and count the specifiers instead of just passing it along blindly. Neither downside would have to exist if c++ just allowed arguments to consteval functions to be treated as compile time constants, which they are by definition, but they did not do this (I think because stylistically they want to see compile-time arguments passed as `<>` arguments not `()` arguments.)

Not sure about any of this, though, would need to experiment.